test/integration/autolinker_spec.mjs

```diff
@@ -121,4 +132,50 @@ describe("autolinker", function () {
       );
     });
   });
+
+  describe("PR 19470", function () {
+    let pages;
+
+    beforeAll(async () => {
+      pages = await loadAndWait(
+        "bug1019475_2.pdf",
+        ".annotationLayer",
+        null,
+        null,
+        {
+          enableAutoLinking: true,
+        }
+      );
+    });
+
+    afterAll(async () => {
+      await closePages(pages);
+    });
+
+    it("must not repeatedly add link annotations redundantly", async () => {
+      await Promise.all(
+        pages.map(async ([browserName, page]) => {
+          await waitForLinkAnnotations(page);
+          let url = await page.$$eval(
+            ".annotationLayer > .linkAnnotation > a",
+            annotations => annotations.map(a => a.href)
+          );
+          expect(url.length).withContext(`In ${browserName}`).toEqual(1);
+
+          await page.evaluate(() =>
+            window.PDFViewerApplication.pdfViewer.updateScale({
+              drawingDelay: -1,
+              scaleFactor: 2,
+            })
+          );
+          await waitForLinkAnnotations(page);
```

Comment:
Does this actually work? If there are no link annotations added (because we already added them), the event is not going to be fired, right?

Should we maybe add a page after the one with the links, and wait for that page after to trigger `pagerendered`? (since it will only be rendered once the first page is done rendering all the layers)

Reply:
The event *should* still fire (in the same the e.g. `annotationlayerrendered`, and many others, are fired when the annotationLayer has been zoomed).